client/src/partials/cost_center/center/analysis_center.js

```diff
@@ -1,122 +1,111 @@
 angular.module('bhima.controllers')
-.controller('analysisCenter', [
-  '$scope',
-  'connect',
-  'appstate',
-  'messenger',
-  'validate',
-  '$translate',
-  'SessionService',
-  function ($scope, connect, appstate, messenger, validate, $translate, SessionService) {
-
-    var dependencies = {};
-
-    dependencies.cost_centers = {
-      query : {
-        tables : {
-          'cost_center' : {
-            columns : ['id', 'text', 'note', 'is_principal']
-          }
+.controller('AnalysisCostCenterController', AnalysisCostCenterController);
+
+AnalysisCostCenterController.$inject = [
+  'connect', 'messenger', 'validate', '$translate', 'SessionService'
+];
+
+/**
+  * Analysis Cost Center Controller
+  * This controller is responsible for make crud operations with cost centers
+  */
+function AnalysisCostCenterController (connect, messenger, validate, $translate, SessionService) {
+  var vm = this,
+      dependencies = {},
+      session = vm.session = {};
+
+  dependencies.cost_centers = {
+    query : {
+      tables : {
+        'cost_center' : {
+          columns : ['id', 'text', 'note', 'is_principal']
         }
       }
-    };
-
-    $scope.register = {};
-    $scope.selected = {};
-
-    startup();
-
-    function startup() {
-      $scope.enterprise = SessionService.enterprise;
-      validate.process(dependencies).then(init);
-    }
-
-    function init(model) {
-      $scope.model = model;
-    }
-
-    function setAction(value, cost_center) {
-      $scope.action = value;
-      $scope.selected = angular.copy(cost_center) || {};
-      $scope.selected.is_principal = $scope.selected.is_principal !== 0;
     }
+  };
 
-    function writeCenter() {
-      return connect.post('cost_center', connect.clean($scope.register));
-    }
+  vm.register = {};
+  vm.selected = {};
 
-    function saveRegistration() {
-      $scope.register.is_principal = ($scope.register.is_principal)? 1 : 0;
-      writeCenter()
-      .then(function() {
-        validate.refresh(dependencies, ['cost_centers']).then(function (model) {
-          angular.extend($scope, model);
-        });
+  startup();
 
-        $scope.register = {};
+  function startup() {
+    session.state = 'loading';
+    vm.enterprise = SessionService.enterprise;
+    validate.process(dependencies).then(init);
+  }
 
-        messenger.success($translate.instant('ANALYSIS_CENTER.INSERT_SUCCESS_MESSAGE'));
-      })
-      .catch(function () {
-        messenger.danger($translate.instant('ANALYSIS_CENTER.INSERT_FAIL_MESSAGE'));
-      });
-    }
+  function init(model) {
+    vm.model = model;
+    session.state = 'loaded';
+  }
 
-    /*
-    function getAvailablesAccounts(enterprise_id) {
-      return connect.fetch('/availablechargeAccounts/'+$scope.project.enterprise_id+'/');
-    }
+  function setAction(value, cost_center) {
+    vm.action = value;
+    vm.selected = angular.copy(cost_center) || {};
+    vm.selected.is_principal = vm.selected.is_principal !== 0;
```

Comment:
I know you didn't write this code, but is there anyway to make sure we are dealing with either
 1. only booleans (true/false)
 2. only integers (1/0)
It would make the code much clearer.  If `is_principal` is bound to a checkbox/radio button, look into using `ng-true-value` and `ng-false-value`.